### PR TITLE
Rework Sidebar's "Load" handling

### PR DESCRIPTION
**Pull request by Joacchim**
See commit messages for details

### CODE REVIEW

File: web-extension/background-engine.js [hunk 1]
```diff
@@ -1,14 +1,52 @@
...
     LOGS:readable
 */
 
+/*
+ * /!\  IMPORTANT - PLEASE READ  /!\
+ *
+ * There are many reasons in this web-extension for the existence of the
+ * "background" or "extension" page (which uses this script as an entry-point).
+ *
+ * First of all, may contraints are pushed onto the web-extension development
```
Comment (GuillaumeGomez) [inline web-extension/background-engine.js:14]:
```suggestion
 * First of all, many contraints are pushed onto the web-extension development
```

File: web-extension/background-engine.js [hunk 2]
```diff
@@ -17,19 +55,28 @@ bmcMessaging.addWindowHandler(
     evData => evData.type === 'computation'
               && evData.module === 'sources'
               && evData.computation === 'URL:Generate:Request',
-    evData => { // Don't receive `sender`, as we don't use it.
+    (evData, sender, sendResponse) => { // Don't receive `sender`, as we don't use it.
```
Comment (GuillaumeGomez) [inline web-extension/background-engine.js:58]:
```suggestion
    (evData, sender, sendResponse) => {
```

File: web-extension/background-engine.js [hunk 3]
```diff
@@ -17,19 +55,28 @@ bmcMessaging.addWindowHandler(
...
-                comic: evData.resource.comic,
-                url: bmcSources.computeURL(evData.resource.reader,
-                                           evData.resource.comic),
+        bmcData.getComic(evData.resource.id, (err, comic) => {
+            let answerEv = {
+                type: 'computation',
+                module: 'sources',
+                computation: 'URL:Generate:Response',
+                resource: {
+                    err: false,
```
Comment (GuillaumeGomez) [inline web-extension/background-engine.js:66]: Why providing the fields if they are at `false` and `null`?

File: web-extension/scripts/load-bookmarks.js [hunk 4]
```diff
@@ -78,21 +78,46 @@ BmcMangaList.prototype.onSourceClick = function(comic, source) {
...
+     *    to load the requested one.
+     *
+     */
     compat.sendMessage(ev, (err, response) => {
         if (err) {
             LOGS.warn('E0013', {'err': err});
             return undefined;
         }
+        if (response.err) {
+            LOGS.warn('E0014', {'err': response.err});
```
Comment (GuillaumeGomez) [inline web-extension/scripts/load-bookmarks.js:119]: Shouldn't it return here?

File: web-extension/background-engine.js [hunk 5]
```diff
@@ -17,19 +55,28 @@ bmcMessaging.addWindowHandler(
...
+                    url: null,
+                },
+            };
+            if (err) {
+                answerEv.err = 'Failed to get Comic data from storage';
```
Comment (GuillaumeGomez) [inline web-extension/background-engine.js:71]: I just comment on it but no need to pay attention: you didn't use translated string stuff. But considering that we want to re-write it, I guess it's fine...

File: web-extension/background-engine.js [hunk 6]
```diff
@@ -17,19 +55,28 @@ bmcMessaging.addWindowHandler(
...
+                return sendResponse(answerEv);
+            }
+            answerEv.resource.url = bmcSources.computeURL(evData.resource.reader, comic.toInfo(evData.resource.reader));
+            if (!answerEv.resource.url) {
+                answerEv.err = 'Could not resolve URL from comic ' + JSON.stringify(comic.toInfo(evData.resource.reader));
```
Comment (GuillaumeGomez) [inline web-extension/background-engine.js:76]: Same.

File: web-extension/background-engine.js (hunk 3 above)
Comment (Joacchim) [reply web-extension/background-engine.js:66]: simply setting the common object's structure, for consistency.
Is that a problem for you ?

File: web-extension/scripts/load-bookmarks.js (hunk 4 above)
Comment (Joacchim) [reply web-extension/scripts/load-bookmarks.js:119]: good point.

File: web-extension/background-engine.js (hunk 5 above)
Comment (Joacchim) [reply web-extension/background-engine.js:71]: Nah, you're right, I'll make it fit the current model, I'd rather have something that we want to change and is consistent, than create inconsistency.

File: web-extension/background-engine.js (hunk 3 above)
Comment (GuillaumeGomez) [reply web-extension/background-engine.js:66]: No, just wondered.

File: web-extension/background-engine.js [hunk 7]
```diff
@@ -50,34 +50,36 @@ const bmcData = new BmcDataAPI();
...
-                resource: {
-                    err: false,
-                    url: null,
-                },
-            };
-            if (err) {
-                answerEv.err = LOGS.getString('S1');
+        return sendResponse => {
+            bmcData.getComic(evData.resource.id, (err, comic) => {
+                let answerEv = {
```
Comment (GuillaumeGomez) [inline web-extension/background-engine.js:62]: Could be `const` as well.

**Review by GuillaumeGomez — APPROVED** [review]

**Pull request merged**